src/QSymbolicsBase/objects.jl

```diff
@@ -0,0 +1,71 @@
+"""This file defines quantum objects (kets, bras, and operators) with various properties"""
+struct SKet <: Symbolic{AbstractKet}
+    name::Symbol
+    basis::Basis
+end
+struct SBra <: Symbolic{AbstractBra}
+    name::Symbol
+    basis::Basis
+end
+struct SOperator <: Symbolic{AbstractOperator}
+    name::Symbol
+    basis::Basis
+end
+const SymQ = Union{SKet, SBra, SOperator}
+isexpr(::SymQ) = false
+metadata(::SymQ) = nothing
+basis(x::SymQ) = x.basis
+
+symbollabel(x::SymQ) = x.name
+Base.show(io::IO, x::SKet) = print(io, "|$(symbollabel(x))⟩")
+Base.show(io::IO, x::SBra) = print(io, "⟨$(symbollabel(x))|")
+Base.show(io::IO, x::SOperator) = print(io, "$(symbollabel(x))")
+Base.show(io::IO, x::SymQObj) = print(io, symbollabel(x)) # fallback that probably is not great
+
+ishermitian(x::SOperator) = false
+isunitary(x::SOperator) = false
+iscommutative(x::SOperator) = false
+
+"""Inverse Operator"""
+@withmetadata struct SInverseOperator <: Symbolic{AbstractOperator}
+    op::Symbolic{AbstractOperator}
+end
+isexpr(::SInverseOperator) = true
+iscall(::SInverseOperator) = true
+arguments(x::SInverseOperator) = [x.op]
+operation(x::SInverseOperator) = inverse
```

Comment:
What is `inverse`? Is such a function declared anywhere?

If `inverse` does not exist we should probably figure out what tests to add so that issues like this one get caught automatically in the future.

Reply:
on reading the rest of the PR, I think I am getting a better idea of what this is (not so much for the other 3). This does not seem like a basic literal. Probably it should go wherever the dagger is defined.